internal/ir/ratelimit_infra.go

```diff
@@ -8,16 +8,6 @@ package ir
 // RateLimitInfra defines managed rate limit service infrastructure.
 // +k8s:deepcopy-gen=true
 type RateLimitInfra struct {
-	// ServiceConfigs for Rate limit service configuration.
-	ServiceConfigs []*RateLimitServiceConfig
-}
-
-// RateLimitServiceConfig holds the rate limit service configurations
-// defined here https://github.com/envoyproxy/ratelimit#configuration-1
-// +k8s:deepcopy-gen=true
-type RateLimitServiceConfig struct {
-	// Name of the config file.
-	Name string
-	// Config contents saved as a YAML string.
-	Config string
+	// ServiceNames for Rate limit service configuration name.
+	ServiceNames []string
```

Comment:
can you elaborate why this is needed ?

Reply:
since we don't use RateLimitServiceConfig anymore, I rm it.

https://github.com/envoyproxy/gateway/blob/334422166f208a03440fcec1534c8fe4a2aab3ef/internal/ir/ratelimit_infra.go#L18-L23

but we do need to keep RateLimitInfra, 

https://github.com/envoyproxy/gateway/blob/334422166f208a03440fcec1534c8fe4a2aab3ef/internal/ir/ratelimit_infra.go#L10-L13

so when we Subscribe the changes from xdsIR channel, like someone enable the global ratelimit:

https://github.com/envoyproxy/gateway/blob/334422166f208a03440fcec1534c8fe4a2aab3ef/internal/globalratelimit/runner/runner.go#L46

the RateLimitInfraIR should be informed, but how can we tell whether is delete/create? so I use `ServiceNames []string`, it's just derived from `[]*RateLimitServiceConfig` but only contains `name`.

```go
if rlIR == nil {
	r.RateLimitInfraIR.Delete(r.Name())
} else {
	r.RateLimitInfraIR.Store(r.Name(), rlIR)
}
```

and infra Subscribe to the RateLimitInfraIR, so it can create/delete the service and deployment of ratelimit.

https://github.com/envoyproxy/gateway/blob/334422166f208a03440fcec1534c8fe4a2aab3ef/internal/infrastructure/runner/runner.go#L76-L79

Reply:
ah thanks for flagging this, I think more deletions are needed :)
* we need to get rid of https://github.com/envoyproxy/gateway/blob/main/internal/ir/ratelimit_infra.go since there is no useful info being passed around from the rl-service to the infra-manager
* we need to get rid of the message https://github.com/envoyproxy/gateway/blob/554176b8cbbc5ffa627dc7109e1b3d2b59474e99/internal/message/types.go#L72 since its no longer useful
* the subscriber in the infra-manager needs to be deleted
https://github.com/envoyproxy/gateway/blob/554176b8cbbc5ffa627dc7109e1b3d2b59474e99/internal/infrastructure/runner/runner.go#L76 since its no longer useful
* the infra manager should deploy a RL service based on whether rate limit is configured or not
https://github.com/envoyproxy/gateway/blob/554176b8cbbc5ffa627dc7109e1b3d2b59474e99/internal/infrastructure/runner/runner.go#L48
it doesnt need to be dynamic

All these pieces can be handled in this PR or a follow up PR is fine, dont want to block this PR, since its tremendous improves the time for ratelimit configuration

Reply:
sure, the more deletions will be done as follow-up

Reply:
resolved with https://github.com/envoyproxy/gateway/pull/1451